lib/swarm.js

```diff
@@ -1,165 +0,0 @@
-var events = require('events')
-var subleveldown = require('subleveldown')
-var swarm = require('webrtc-swarm')
-var signalhub = require('signalhub')
-var hyperlog = require('hyperlog')
-var ghsign = require('ghsign')
-var through = require('through2')
-var protobuf = require('protocol-buffers')
-var fs = require('fs')
-var request = require('request')
-var wrtc = require('wrtc')
-
-var messages = protobuf(fs.readFileSync(__dirname + '/../schema.proto'))
-
-module.exports = createSwarm
-
-ghsign = ghsign(function (username, cb) {
-  fs.readFile('./public-keys/' + username + '.keys', 'utf-8', function (_, keys) {
-    if (keys) return cb(null, keys)
-    request('https://github.com/' + username + '.keys', function (_, response) {
-      var keys = response.statusCode === 200 && response.body
-      if (!keys) return cb(new Error('Could not find public keys for ' + username))
-      fs.mkdir('./public-keys', function () {
-        fs.writeFile('./public-keys/' + username + '.keys', keys, function () {
-          cb(null, keys)
-        })
-      })
-    })
-  })
-})
-
-var verifiers = {}
-
-function createSwarm (db) {
-  var emitter = new events.EventEmitter()
-  var logs = {}
-
-  emitter.logs = logs
-
-  var sign
-  var processor
-
-  emitter.changes = function (name) {
-    return logs[name] ? logs[name].changes : 0
-  }
-
-  emitter.process = function (fn) {
-    processor = fn
-    Object.keys(logs).forEach(function (name) {
-      startProcessor(logs[name])
-    })
-  }
-
-  emitter.removeChannel = function (name) {
-    var log = logs[name]
-    if (!log) return
-    delete logs[name]
-
-    if (log.processing) log.processing.destroy()
-    log.peers.forEach(function (p) {
-      p.destroy()
-    })
-  }
-
-  emitter.addChannel = function (name) {
-    if (logs[name]) return
-
-    var log = logs[name] = hyperlog(subleveldown(db, name))
-    var id = 'friends-' + name
-    var hub = signalhub('http://dev.mathiasbuus.eu:8080', id)
-    var sw = swarm(hub, { wrtc: wrtc })
```

Comment:
Since friends-irc uses node-webrtc wrtc was passed in as an option here. Should be good if you just change the call to `createSwarm` [here](https://github.com/moose-team/friends-irc/blob/friends-swarm/lib/bot.js#L46) to pass in the options instead :)